Shared/LayerCacheManager.cpp

```diff
@@ -365,6 +319,117 @@ void LayerCacheManager::addExcludedPath(const QString& exludedPath)
+{
+  const QVariant elevationData = properties.value(ELEVATION_PROPERTYNAME);
+  const QStringList pathList = elevationData.toStringList();
+
+  // If size is 1, it could be a TPK or raster
+  if (pathList.length() == 1)
+  {
+    // Get the string
+    const QString elevationSource = pathList.at(0);
+    QFileInfo elevationSourceInfo(elevationSource);
+
+    // Check if TPK or not
+    if (elevationSourceInfo.suffix().toLower() == "tpk")
+      m_localDataController->createElevationSourceFromTpk(elevationSource);
+    else
+      m_localDataController->createElevationSourceFromRasters(QStringList{elevationSource});
+  }
+  // If more than 1, it is a list of rasters
+  else if (pathList.length() > 1)
+  {
+    m_localDataController->createElevationSourceFromRasters(pathList);
+  }
+}
+
+void LayerCacheManager::addLayers(const QVariantMap& properties)
+{
+  const QVariant layersData = properties.value(LAYERS_PROPERTYNAME);
+  const auto layersList = layersData.toList();
+  m_inputLayerJsonArray = QJsonArray::fromVariantList(layersList);
+
+  auto it = m_inputLayerJsonArray.constBegin();
+  auto itEnd = m_inputLayerJsonArray.constEnd();
+  int layerIndex = 0;
+  for (; it != itEnd; ++it)
+  {
+    const QJsonValue jsonVal = *it;
+    if (jsonVal.isNull())
+      continue;
+
+    const QJsonObject jsonObject = jsonVal.toObject();
+    if (jsonObject.isEmpty())
+      continue;
+
+    jsonToLayer(jsonObject, layerIndex);
+
+    layerIndex++;
+  };
+}
+
+void LayerCacheManager::connectSignals()
+{
+  // obtain Scene and connect slot
+  m_scene = Toolkit::ToolResourceProvider::instance()->scene();
+  if (m_scene)
+  {
+    // disconnect signals
+    if (m_dataChangedConnection)
+      disconnect(m_dataChangedConnection);
+
+    if (m_layerAddedConnection)
+      disconnect(m_layerAddedConnection);
+
+    if (m_layerRemovedConnection)
+      disconnect(m_layerRemovedConnection);
+
+    if (m_layoutChangedConnection)
+      disconnect(m_layoutChangedConnection);
+
+    if (m_modelResetConnection)
+      disconnect(m_modelResetConnection);
+
+    // connect signals
+    m_dataChangedConnection = connect(m_scene->operationalLayers(), &LayerListModel::dataChanged, this, &LayerCacheManager::onLayerListChanged); // layer objects have been added or changed
+    m_layerAddedConnection = connect(m_scene->operationalLayers(), &LayerListModel::layerAdded, this, &LayerCacheManager::onLayerListChanged); // layer objects have been added
+    m_layerRemovedConnection = connect(m_scene->operationalLayers(), &LayerListModel::layerRemoved, this, &LayerCacheManager::onLayerListChanged); // layer has been removed
+    m_layoutChangedConnection = connect(m_scene->operationalLayers(), &LayerListModel::layoutChanged, this, &LayerCacheManager::onLayerListChanged); // order changed
+    m_modelResetConnection = connect(m_scene->operationalLayers(), &LayerListModel::modelReset, this, &LayerCacheManager::onLayerListChanged); // order changed
+  }
+
+  if (m_localDataController)
```

Comment:
This is optional but if you consider inverting this check, along with `if (m_scene)` above, you can reduce the nesting.

For example
```c++
if (m_scene)
{
  // lots of logic
  if (m_localDataController)
  {
    // lots of logic
  }
}
```
to
```c++
if (!m_scene)
  return;

// lots of logic

if (!m_localDataController)
  return;

// lots of logic
```